spec/unit/outputs/elasticsearch/http_client_spec.rb

```diff
@@ -180,6 +180,53 @@
     end
   end
 
+  describe "#bulk" do
+    subject { described_class.new(base_options) }
+
+    require "json"
+    let(:message) { "hey" }
+    let(:actions) { [
+      ["index", {:_id=>nil, :_index=>"logstash"}, {"message"=> message}],
+    ]}
+
+    context "if a message is over TARGET_BULK_BYTES" do
+      let(:target_bulk_bytes) { LogStash::Outputs::ElasticSearch::TARGET_BULK_BYTES }
+      let(:message) { "a" * (target_bulk_bytes + 1) }
+
+      it "should be handled properly" do
+        allow(subject).to receive(:join_bulk_responses)
+        expect(subject).to receive(:bulk_send).once do |data|
+          expect(data.size).to be > target_bulk_bytes
+        end
+        s = subject.send(:bulk, actions)
+      end
+    end
+
+    context "with two messages" do
+      let(:message1) { "hey" }
+      let(:message2) { "you" }
+      let(:actions) { [
+        ["index", {:_id=>nil, :_index=>"logstash"}, {"message"=> message1}],
+        ["index", {:_id=>nil, :_index=>"logstash"}, {"message"=> message2}],
+      ]}
+      it "executes one bulk_send operation" do
```

Comment:
nitpick: adding a line of whitespace between the setup `let` statements and the specs themselves would be consistent with our other example groups, and add a bit of visual separation making it easier to skim.